prometheus/doc/exporter-for-other-services.md

```diff
@@ -148,4 +148,57 @@ If service decide to leverage node-exporter to expose metrics, service should mo
 located in `/datastorage/prometheus` and write file with extension `.prom`.
 
 For expose using port, service should annotate pod with `prometheus.io/scrape`, `prometheus.io/path`
-and `prometheus.io/port`.
+and `prometheus.io/port`. For service with label key `app`, prometheus will generate `pai_service_name`
+label for generated metrics.
+
+prometheus will list all pods and get those pods with `prometheus.io/scrape` is true, and then,
+prometheus will issue request to `http://${pod_host_ip}:${prometheus.io/port}${prometheus.io/path}` to
+get metrics.
+
+For example, following deployment:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: foo
+spec:
+  selector:
+    matchLabels:
+      app: foo
+  template:
+    metadata:
+      name: foo
+      labels:
+        app: foo
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: "/"
+        prometheus.io/port: "80"
+    spec:
+      containers:
+      - name: nginx
```

Comment:
Question:
1 nginx is sidecar container for other services to expose metrics at specific port? if this , it's better tell user this info 
2 Or nginx is a exmple for nginx service expose metrics?
3 If suggest to use port to expose, how about tell owners to Unified definition of port usage to avoid conlict

Reply:
This is a simple example to use port to expose something. I can update doc to illustrate it more clearly.

I really do not know how to avoid port conflict. Maybe coordinate with other to use different port. The best solution for this is enable kubernetes network model and make service use private network. I will update doc to suggest coordinate with others.